pkg/stats/monitor.go

```diff
@@ -146,17 +151,15 @@ func (m *Monitor) GetCPULoad() float64 {
 func (m *Monitor) CanAcceptRequest(req *rpc.StartEgressRequest) bool {
 	accept := false
 
-	total := float64(m.cpuStats.NumCPU())
-	available := m.cpuStats.GetCPUIdle() - m.pendingCPUs.Load()
+	numCPU := float64(m.cpuStats.NumCPU())
 
-	if total-available <= 0.01 {
-		// if idle, add room for when cost == numCPUs
-		available += 0.01
-	} else {
-		// if not idle, leave 20% CPU
-		available -= total * 0.2
+	count := m.egressCount.Load()
+	reserved := m.pendingCPUs.Load()
+	if count > 0 {
+		reserved += (numCPU - m.cpuStats.GetCPUIdle()) * math.Pow(breathingRoom, float64(count))
```

Comment:
what's the intention of this? I'm plugging some numbers and it doesn't seem to produce sane values for me. can we add tests for this logic to demonstrate the expected & edge cases?

i.e. 8 cpus, with 4 idle and 2 current egresses.. the logic becomes:

(8-4) * 1.2^2 = 5.76 reserved

also should all egress types be treated the same? what happens when I have 20 track egresses that consume no CPU.. wouldn't the reserved count be out of wack?

Reply:
the idea was the more egresses that are running, the more of a risk it is to add another. So 8 cpus with 4 idle and 2 egresses -> 5.76 reserved, 3.24 remaining, if cpu cost is 3 it would accept a third

8 cpus with 3 idle and 5 egresses -> (8-3) * 1.2^5 = 12.4, and it wouldn't accept a 6th